### DISCUSSION TITLE
Password Reset

### DISCUSSION THREAD

**Discussion by ghost** [louislam/dockge#194] [Ask for Help] [answered]
### ⚠️ Please verify that this bug has NOT been raised before.

- [X] I checked and didn't find similar issue

### 🛡️ Security Policy

- [X] I agree to have read this project [Security Policy](https://github.com/louislam/dockge/security/policy)

### 📝 Describe your problem

Is there a way to reset the password like in uptime kuma ? Didnt find anything in package.json

### 🐻 Dockge Version

1

### 💻 Operating System and Arch

20.04.6

### 🌐 Browser

Chrome

### 🐋 Docker Version

24.0.7

**Comment by louislam** [comment]
Good point, I forgot to copy this feature from Uptime Kuma.

Workaround:
Since most features currently is depending on the "stacks". You can:
1. Stop Dockge
2. Delete your Dockge's data directory
3. Start Dockge and create your admin account again. You won't lose any stacks.

**Reply by ghost** [reply]
Yes exactly i ended up doing that, thank you !

**Comment by breeegz** [comment]
> Workaround: Since most features currently is depending on the "stacks". You can:
> 
> 1. Stop Dockge
> 2. Delete your Dockge's data directory
> 3. Start Dockge and create your admin account again. You won't lose any stacks.

I tried to do that, my admin account was recreated, but I can't control my stacks (even after rescanning my Stack-Folder)

All my stacks are saying the same message on the WebGui,  "This stack is not managed by Dockge"

Fortunately, I moved my data folder instead of deleting it, although I'm not sure what good it's going to do me without the admin password.

**Reply by louislam** [reply, marked as answer]
The answer about is outdated. Now you can reset your password by this command:

```
cd /opt/dockge
docker compose exec dockge pnpm run reset-password
```

**Reply by edoelas** [reply]
Is there a way to specify the username and password from the CLI?

**Reply by danmcmullen** [reply]
I would like to know this as well?

**Reply by Pr0mises** [reply]
That would be awesome

**Reply by cybersholt** [reply]
Given the complexity that can arise when trying to reset passwords, I believe it would be highly beneficial for users if the password reset instructions provided in this discussion were included in the README file. 

This would offer easier access and visibility for all users needing to perform this common task.

**Reply by breeegz** [reply]
```bash
cloudvm:/opt/dockge$ docker compose exec dockge pnpm run reset-password
 WARN  Unsupported engine: wanted: {"node":">= 18.0.0 && <= 18.17.1"} (current: {"node":"v18.17.1","pnpm":"8.11.0"})

> dockge@1.4.2 reset-password /app
> tsx ./extra/reset-password.ts

== Dockge Reset Password Tool ==
2024-04-23T00:50:34Z [SERVER] INFO: NODE_ENV: production
2024-04-23T00:50:34Z [SERVER] INFO: Server Type: HTTP
Connecting the database
2024-04-23T00:50:34Z [DB] INFO: Database Type: sqlite
2024-04-23T00:50:34Z [SERVER] INFO: Connected to the database
Found user: existinguser1
New Password: newsecurepassword
Confirm New Password: newsecurepassword
Password reset successfully.
Connecting to ws://localhost:5001 to disconnect all other socket clients
Logged in.
2024-04-23T00:51:08Z [DB] INFO: Closing the database
2024-04-23T00:51:10Z [DB] INFO: Database closed
Finished.
```

This was great!  Much better than the workaround from my comment in December, which was probably my own fault.

**Reply by frclaude** [reply]
I am trying to reset password, got this : 
`

\> dockge@1.5.0 reset-password
\> tsx ./extra/reset-password.ts

== Dockge Reset Password Tool ==
2025-10-24T15:14:24Z [SERVER] INFO: NODE_ENV: production
2025-10-24T15:14:24Z [SERVER] INFO: Server Type: HTTP
Connecting the database
2025-10-24T15:14:24Z [DB] INFO: Database Type: sqlite
2025-10-24T15:14:24Z [SERVER] INFO: Connected to the database
Found user: cej
New Password: ******
Confirm New Password: ******
Password reset successfully.
Connecting to ws://localhost:5001 to disconnect all other socket clients
Logged in.
2025-10-24T15:14:29Z [DB] INFO: Closing the database
2025-10-24T15:14:31Z [DB] INFO: Database closed
Finished.

`

Tryed several times, but I still cannot log in....

**Comment by laurentlemercier** [comment]
Pull request provided in order to handle this :  Allow to set password using an environment variable at runtime #426

**Comment by stueck9356** [comment]
This is mostly for me in the future, however for those running dockge in Truenas Scale, do the following:

1. Go to the apps tab, select dockge and open the shell for the container
![image](https://github.com/user-attachments/assets/05607e0b-6204-44ea-b122-e042901d4709)
2. Type `cd /app/extra`, hit enter
4. Type `docker exec -it ix-dockge-dockge-1 pnpm run reset-password`, hit enter

if you get a "No such container" message, you will need to
1. type `docker ps`, hit enter
3. Find the image for dockge and copy the NAMES
![image](https://github.com/user-attachments/assets/f64ce793-7d93-447d-99a6-1c78356c089e)
4. Type 'docker exec -it <"NAMES"> pnpm run reset-password', hit enter

**Reply by tknospdr** [reply]
I tried your command and got this:
`OCI runtime exec failed: exec failed: unable to start container process: exec: "pnpm": executable file not found in $PATH: unknown`

**Reply by cyril59310** [reply]
Dockge has switched back to npm, so you’ll need to use `docker exec -it dockge npm run reset-password`

**Reply by jaeti01** [reply]
I wonder if this is still working?
When I try it, I get the following output:

```
root@dockerenv:/opt/dockge# docker exec dockge-dockge-1 npm run reset-password

> dockge@1.5.0 reset-password
> tsx ./extra/reset-password.ts

== Dockge Reset Password Tool ==
2025-05-28T14:52:51Z [SERVER] INFO: NODE_ENV: production
2025-05-28T14:52:51Z [SERVER] INFO: Server Type: HTTP
Connecting the database
2025-05-28T14:52:51Z [DB] INFO: Database Type: sqlite
2025-05-28T14:52:51Z [SERVER] INFO: Connected to the database
Found user: admin
Error: readline was closed
2025-05-28T14:52:51Z [DB] INFO: Closing the database
2025-05-28T14:52:53Z [DB] INFO: Database closed
Finished.
```
But password didn't seem to be reset afterwards.

**Reply by cyril59310** [reply]
@jaeti01 I’m not sure if you’ve resolved your issue, but this should help you

You need to connect directly to the container using:

`docker exec -it <container_name> bash`
Then, run the following command:

`npm run reset-password`

**Reply by jaeti01** [reply]
Thanks for your reply. I tried that as well and it didn't work either. But the problem is more or less not there anymore because I managed to "brute-force" the password and change it through the UI.

**Reply by vazquezjm** [reply]
> @jaeti01 I’m not sure if you’ve resolved your issue, but this should help you
> 
> You need to connect directly to the container using:
> 
> `docker exec -it <container_name> bash` Then, run the following command:
> 
> `npm run reset-password`

This worked for me. Thanks!

**Comment by danmcmullen** [comment]
***@***.***


*Dan McMullen*
c 902.449.1773

**Comment by mitchsurp** [comment]
Answering this from 2026 as the previous answers didn't work for me. I was probably doing something wrong but npm wasn't playing nice. 

 I had to sh into the container

`docker compose exec dockge sh`

And then

`# cd /app && node_modules/.bin/tsx extra/reset-password.ts`
